### DISCUSSION TITLE
Adding LTI integration to self-hosted production servers

### DISCUSSION THREAD

**Discussion by myriath** [PrairieLearn/PrairieLearn#9626] [Q&A] [answered]
Hello, I am currently trying to link my PrairieLearn course into my University's Canvas page, and running into a few issues.

First, the LTI page of the documentation says to use the Admin / LTI page for the course instance, but the page there says it is deprecated. I am having trouble finding documentation for the replacement, which seems to be LTI 1.3 as suggested by other issues and discussions, but I am not sure.

It seems like 1.3 might not be implemented yet, but the discussion I found for that was back in 2022, so is this still the case?
If 1.1 is the only currently supported way to link PrairieLearn, how might I go about enabling it for my production server and accessing the settings? I couldn't seem to find this in the docs anywhere.

Thank you for the help!

**Comment by nwalters512** [comment, marked as answer]
LTI 1.1 is indeed deprecated and no longer recommended for use. Our LTI 1.3 functionality is in active development but is currently an enterprise-only feature. If you'd like to discuss an enterprise license for a self-hosted installation, you can reach out via our contact form at https://www.prairielearn.com/contact.

**Reply by myriath** [reply]
So does this mean that there is no currently supported way to integrate LTI without a license?

**Reply by nwalters512** [reply]
That's correct. You can see availability details about our various features on our pricing page: https://www.prairielearn.com/pricing

**Reply by myriath** [reply]
Is there a timeline for removal of LTI 1.1? I set up a test on an empty Canvas course and it seems to do exactly what I need it to, but I am worried it's going to be removed soon.

The issue for 1.3 is needing to go through specific channels in my university, while 1.1 seems to just work. Or, at least, I can't actually set up 1.3 apps in the basic Canvas environment I have for testing it.

Thanks for your help with this

**Reply by nwalters512** [reply]
We don't have a concrete timeline for removal, but we won't do so until 1.3 has support for grade passback.

I should ask to clarify: are you in fact self-hosting PrairieLearn? You mentioned "my production server" so I assumed you were. If that is the case, you're always free to maintain a version that supports LTI 1.1 indefinitely.

The main problem with LTI 1.1 is that it intentionally doesn't provide student identifying information to tool platforms; you'd probably be able to see that from your quick test. This makes it difficult to do course admin work, e.g. looking up specific students by name or email. If you're comfortable with that constraint, LTI 1.1 might be workable for you for now!

**Reply by myriath** [reply]
Yes, I am self-hosting, so that makes sense, I was just worried about having to maintain it on my own. Good to know that it will be around for at least a little while in the official repository.

Thank you for your help, I think I will be sticking with 1.1, at least until I can figure out a way to get 1.3 to work with my university's setup.

**Reply by nwalters512** [reply]
Happy to help! If/when you're ready to explore LTI 1.3 on our hosted platform, please reach out, we're very comfortable working with institutions to go through that approval and configuration process (we haven't seen an institution yet with self-service Canvas app configuration). Or if you'd like to continue self-hosting, we can discuss a self-hosted license as mentioned above.

**Comment by mfsilva22** [comment]
Hi, I think we met at SIGCSE. Nice to see you around!
You can add the PrairieLearn assessment links to your Canvas page, so students can still go to your LMS and then get directed to PrairieLearn. Many instructors at UIUC and other universities use this approach. Then you can download the grades from PrairieLearn and upload them to Canvas. You will not need a license (EE features) to use this approach.

**Reply by myriath** [reply]
Thank you for the suggestion. Yeah I realize that that is a solution and would be simpler to set up, I'm just currently looking for a way to reduce the amount of work TAs / professors need to do to get the grades into Canvas.

I'm not sure if I've ever heard of SIGCSE, so you may have me mistaken for someone else.

**Reply by mfsilva22** [reply]
I am sorry about the confusion! SIGCSE is the ACM Technical Symposium on Computer Science Education that happened last week. I met someone with your name who was asking about guidelines to set up their own server. Happy to help anyway :-)

**Comment by racheldh12** [comment]
Hi Mitch! I actually inherited this project :) Can you walk me through how/if you got 1.1 to work? I have my own Canvas test page, and the PL course instance is giving this: 
![Capture](https://github.com/user-attachments/assets/25584538-08a1-4812-be90-d658150cf0a3)
I want to enable 1.1 on the server and mess around with other ways to speed up the downloading + uploading grades

**Reply by myriath** [reply]
I'm assuming you've already tried the instructions on the docs?  https://prairielearn.readthedocs.io/en/latest/courseInstance/#lti-overview

I can't remember how exactly I got it working but if you have access to the documentation we made last year it should be in there somewhere as a writeup.
